lib/vehicle.rb

```diff
@@ -1,18 +1,23 @@
 require 'date'
 
 class Vehicle
-  attr_reader :vin,
-              :year,
-              :make,
-              :model,
-              :engine
+  attr_reader   :vin,
+                :year,
+                :make,
+                :model,
+                :engine
+
+  attr_accessor :registration_date,
+                :plate_type
 
   def initialize(vehicle_details)
     @vin = vehicle_details[:vin]
     @year = vehicle_details[:year]
     @make = vehicle_details[:make]
     @model = vehicle_details[:model]
     @engine = vehicle_details[:engine]
+    @plate_type = vehicle_details[:plate_type]
```

Comment:
I would not have this here. I would just use your `plate_type` method below, it is more dynamic.